### PR TITLE
Per-Domain Datasets

### PR DESCRIPTION
**Pull request by usert5432**
Hi everyone,

Following our discussion of the Dataset/Dataloader [refactoring](https://github.com/LS4GAN/toygan/issues/63) I have implemented two new toyzero Datasets. These new datasets do not perform any sort of random sampling and therefore are robust when used in a multiprocessing setup. Additionally, instead of returning a pair of `(fake, real)` images they return a single image (either `fake` or `real`).

In addition to implementing the new datasets, this PR adds a few more changes:
* Adds support for a proper train/val/test separation: a5a4ec7a695bcf0b5fa3ee0f46817d8c39356e09, ea8752eb49d4ed705c1b04d443eba01eda724c12. (Previously, separation was supported only for train/test split)
* Modifies dataset viewer script to support new datasets 7d8f18a5aae77aaa4460587459eef2aa594274d6.
* plus a few small documentation, code structure changes.

Inviting: @brettviren, @YHRen, @pphuangyi, @HaiwangYu

### CODE REVIEW

File: toytools/collect.py [hunk 1]
```diff
@@ -120,23 +119,29 @@ def load_image(root : str, is_fake : bool, name : str) -> np.ndarray:
...
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Split dataset of size `n` into training/val/test parts.
 
     Parameters
     ----------
     n : int
         Size of the dataset to split.
+    val_size : int or float
+        Fraction of the dataset that will be used as a val sample.
+        If `test_size` <= 1, then it is treated as a fraction, i.e.
```
Comment (YHRen) [inline toytools/collect.py]: `test_size` -> `val_size`?

**Review by YHRen — APPROVED** [review]
Looks really good. :D

File: toytools/collect.py (hunk 1 above)
Comment (usert5432) [reply toytools/collect.py]: Whoops. Fixed in: [9a7d6e1](https://github.com/LS4GAN/toytools/pull/19/commits/9a7d6e1b3a6f985f3fad42dbe7e545b8e989c858). Thanks!

**Review by YHRen — APPROVED** [review]
nice~

**Review by brettviren — APPROVED** [review]

**Comment by usert5432** [comment]
Thank you for the reviews, @brettviren and @YHRen! I will merge this PR.

**Pull request merged**